File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public int maxDepth(Node root) {
        if(root==null) return 0;
        int ans = 0;
        for(int i = 0; i<root.children.size(); i++){
            int tempAns = maxDepth(root.children.get(i));
            ans = Math.max(ans, tempAns);
        }
        return ans+1;
    }
}

File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number}
 */
var maxDepth = function(root) {
    if(root==null) return 0
    let max = 0
    for(let node of root.children){
        let tempMax = maxDepth(node)
        max = Math.max(max, tempMax)
    }
    return max+1
};